apps/admin-gui/src/app/shared/components/dialogs/applications-list-columns-change-dialog/applications-list-columns-change-dialog.component.ts

```diff
@@ -17,51 +19,76 @@ export interface ApplicationsListColumnsChangeDialogData {
   selector: 'app-applications-list-columns-change-dialog',
   templateUrl: './applications-list-columns-change-dialog.component.html',
   styleUrls: ['./applications-list-columns-change-dialog.component.scss'],
+  providers: [ApplicationColumnSelectLabelPipe],
 })
 export class ApplicationsListColumnsChangeDialogComponent implements OnInit {
   loading = false;
   theme: string;
   simpleColumns = ['createdAt', 'type', 'state', 'createdBy', 'modifiedBy'];
   columnOptions = ['createdAt', 'type', 'state', 'createdBy', 'modifiedBy'];
-  selectedColumns = new FormControl<string[]>([]);
+  selectedColumns: string[] = [];
   attribute: Attribute;
+  templateAttribute: Attribute;
+  fedAttributeDefs: AttributeDefinition[] = [];
+  dataSource: MatTableDataSource<string>;
 
   constructor(
     private dialogRef: MatDialogRef<ApplicationsListColumnsChangeDialogComponent>,
     private attributesManager: AttributesManagerService,
     @Inject(MAT_DIALOG_DATA) private data: ApplicationsListColumnsChangeDialogData,
     private translate: PerunTranslateService,
-    private notificator: NotificatorService
+    private notificator: NotificatorService,
+    public columnNamePipe: ApplicationColumnSelectLabelPipe
   ) {}
 
+  columnTranslation = (name: string): string =>
+    this.columnNamePipe.transform(this.friendlyToDisplayAttrName(name));
+
   ngOnInit(): void {
     this.theme = this.data.theme;
+    this.loading = true;
     if (this.data.groupId) {
```

Comment:
There is a lot of duplication here. Could we extract some parts to make it cleaner ?